suricata/update/commands/checkversions.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2017 Open Information Security Foundation
+#
+# You can copy, redistribute or modify this Program under the terms of
+# the GNU General Public License version 2 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# version 2 along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+from __future__ import print_function
+
+import os.path
+import logging
+from suricata.update import sources
+
+logger = logging.getLogger()
+
+
+def register(parser):
+    parser.set_defaults(func=check_version)
+
+
+def check_version(suricata_version):
+    index_filename = sources.get_index_filename()
+    if not os.path.exists(index_filename):
+        logger.warning("No index exists, will use bundled index.")
+        logger.warning("Please run suricata-update update-sources.")
+    index = sources.Index(index_filename)
+    version = index.get_versions()
+    if suricata_version.full in version['suricata']['recommended'] or \
+            "dev" in suricata_version.full:
```

Comment:
I wonder if we need a bit more logic around 'dev' here. For example, someone may run:
```
4.1.0-dev (git master-4.1.x branch)
5.0.0-dev
4.0.0-dev <- EOL
3.2dev <- EOL
3.1dev <- EOL
```
Note that the 3.x versions use a different syntax.

Reply:
@victorjulien how about modifying index and adding `development: '4.1.0-dev ', '5.0.0-dev'` in it and according adding checks in the code for checking if its a development version? If yes, output is `Suricata-version up to date` else `EOL`. Is this the correct approach?

Reply:
I think perhaps we can do an easier way? If we take the short version (3.1, 4.1, 5.0) and see if:

EOL case:
- if the short version is not defined in the yaml AND
- it is < than the short version of the recommended version -> EOL

Up to date case:
- if the short version is not defined in the yaml AND
- it is >= than the short version of the recommended version -> up2date

Both only for when 'dev' is part of the full version.

Does this make sense?

Reply:
Yes, I understood. Thanks :)